### PR TITLE
schema upgrades bug fixes batch 2/3

### PR DESCRIPTION
**Pull request by manivoxel51**
## What changes are proposed in this pull request?

bug fixes for schema upgrades + UI polish

Screen Shot 2023-05-11 at 8 42 46 AM

Screen Shot 2023-05-11 at 8 42 53 AM

Screen Shot 2023-05-11 at 8 42 40 AM

## How is this patch tested? If it is not, please explain why.

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: app/packages/core/src/components/Sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -22,6 +27,7 @@ import ViewSelection from "./ViewSelection";
 import { resizeHandle } from "./Sidebar.module.css";
 import SchemaSettings from "../Schema/SchemaSettings";
 import { graphql, useFragment } from "react-relay";
+import { selectedPathsState } from "@fiftyone/state/src/hooks/useSchemaSettings";
```
Comment (benjaminpkane) [inline app/packages/core/src/components/Sidebar/Sidebar.tsx:30]: nit: src import

File: app/packages/state/src/hooks/useSchemaSettings.ts [hunk 2]
```diff
@@ -35,14 +37,16 @@ const skipFiled = (path: string, ftype: string) => {
...
   return (
     path === "tags" ||
     path === "filepath" ||
     (ftype === OBJECT_ID_FIELD && path === "id") ||
     path.startsWith("metadata") ||
     path.endsWith("frames.frame_number") ||
-    ftype === FRAME_NUMBER_FIELD
+    ftype === FRAME_NUMBER_FIELD ||
+    path === "group" ||
+    path === "frames.group"
```
Comment (benjaminpkane) [inline app/packages/state/src/hooks/useSchemaSettings.ts:49]: I don't think this actually covers the general case case of group datasets. I suppose dynamic groups (as views) complicates things more. Ideally there is a way to inspect a required group field in the schema cc @brimoor

File: app/packages/state/src/hooks/useSchemaSettings.ts [hunk 3]
```diff
@@ -76,23 +80,31 @@ export const schemaSearchRestuls = atom<string[]>({
...
-            path &&
-            (viewSchema?.[path]?.ftype || schema?.[path]?.ftype) &&
-            !skipFiled(path, viewSchema[path].ftype)
-        );
+        const greenPaths = [...newPaths]
+          .filter(
+            (path) =>
+              path &&
+              (viewSchema?.[
+                path.startsWith("frames.") ? path.replace("frames.", "") : path
```
Comment (benjaminpkane) [inline app/packages/state/src/hooks/useSchemaSettings.ts:88]: This could break image datasets with a `frames` field. There are probably existing issues with a `frames` field on image datasets but just sharing some context. Ok with me until a more formal discussion happens

**Review by benjaminpkane — CHANGES_REQUESTED** [review]
I think group datasets with arbitrary field naming may need to be tested still.
* https://docs.voxel51.com/user_guide/groups.html#creating-grouped-datasets

`dataset.add_group_field()` can take any name for a group field

File: app/packages/state/src/hooks/useSchemaSettings.ts (hunk 2 above)
Comment (manivoxel51) [reply app/packages/state/src/hooks/useSchemaSettings.ts]: Got it fixed thanks to you! it is looking at the dataset.group_field now instead

File: app/packages/state/src/hooks/useSchemaSettings.ts (hunk 3 above)
Comment (manivoxel51) [reply app/packages/state/src/hooks/useSchemaSettings.ts:89]: do you mean custom defined `frames` field on image datasets? videos are the only ones with `frames` prefix I thought 🤔

Comment (benjaminpkane) [reply app/packages/state/src/hooks/useSchemaSettings.ts:90]: I always developed the App to in theory support adding an arbitrary `frames` field, e.g.
```py
dataset.add_sample_field("frames", fo.IntField)
```

But it could very likely not be supported in the SDK / in general. It's just not officially documented. I digress

Comment (brimoor) [reply app/packages/state/src/hooks/useSchemaSettings.ts:90]: The SDK is careful to allow `frames` fields for image datasets. The `frames` keyword only has special meaning for datasets that contain videos.

File: app/packages/state/src/hooks/useSchemaSettings.ts [hunk 4]
```diff
@@ -30,19 +33,22 @@ const skipFiled = (path: string, ftype: string) => {
   return (
     ftype === DICT_FIELD ||
```
Comment (benjaminpkane) [inline app/packages/state/src/hooks/useSchemaSettings.ts:34]: [`fieldsReducer`](https://github.com/voxel51/fiftyone/blob/adc271617adb47e062b41736610d222513126a94/app/packages/state/src/recoil/sidebar.ts#L147) and/or [`UNSUPPORTED_FILTER_TYPES`](https://github.com/voxel51/fiftyone/blob/adc271617adb47e062b41736610d222513126a94/app/packages/utilities/src/index.ts#L318) could be leveraged here

File: app/packages/state/src/hooks/useSchemaSettings.ts [hunk 5]
```diff
@@ -30,19 +33,22 @@ const skipFiled = (path: string, ftype: string) => {
...
     ftype === JUST_FIELD ||
+    ftype === VECTOR_FIELD ||
     path.includes(".logits") ||
```
Comment (benjaminpkane) [inline app/packages/state/src/hooks/useSchemaSettings.ts:37]: `logits` is a vector field so it shouldn't be necessary

File: app/packages/state/src/hooks/useSchemaSettings.ts [hunk 6]
```diff
@@ -30,19 +33,22 @@ const skipFiled = (path: string, ftype: string) => {
...
     path.endsWith(".index") ||
     path.endsWith(".bounding_box")
```
[the remark this answers is not in the document]
Comment (manivoxel51) [reply app/packages/state/src/hooks/useSchemaSettings.ts:39]: to remedy, I added a check to the current's path's parent's embedded document. We skip if it is Detection type and path ends with`.bounding_box`.

File: app/packages/state/src/hooks/useSchemaSettings.ts [hunk 7]
```diff
@@ -26,23 +33,28 @@ export const TAB_OPTIONS_MAP = {
...
     path === "tags" ||
     path === "filepath" ||
     (ftype === OBJECT_ID_FIELD && path === "id") ||
+    ftype === FRAME_NUMBER_FIELD ||
+    groupField === path ||
+    path === "sample_id" ||
     path.startsWith("metadata") ||
-    path.endsWith("frames.frame_number") ||
-    ftype === FRAME_NUMBER_FIELD
+    path.endsWith("frames.frame_number")
```
Comment (benjaminpkane) [inline app/packages/state/src/hooks/useSchemaSettings.ts:57]: Using the [`FrameNumberField`](https://github.com/voxel51/fiftyone/blob/c9f37ad3f8ea52bb69d80e19e466d7cc9591dfb7/fiftyone/core/fields.py#L1033) might be cleaner here

**Review by benjaminpkane — APPROVED** [review]
Code LGTM! 👍

File: app/packages/state/src/hooks/useSchemaSettings.ts (hunk 7 above)
Comment (manivoxel51) [reply app/packages/state/src/hooks/useSchemaSettings.ts:57]: good point. I'll double check in quick followup PR
